src/frontend/components/MainSidebar/MainSidebar.js

```diff
@@ -0,0 +1,86 @@
+import React, { useRef } from "react";
+import { NavLink } from "react-router-dom";
+import { useSidebar } from "../../contexts";
+import { useOnClickOutside, useWindowSize } from "../../hooks";
+import "./MainSidebar.css";
+
+export const MainSidebar = () => {
+  const MainSidebarRef = useRef();
+
+  const sidebarLinkClass = ({ isActive }) =>
+    isActive ? "expanded-sidebar-link active-link" : "expanded-sidebar-link";
+
+  const { setShowSidebar } = useSidebar();
+  const windowSize = useWindowSize();
+
+  useOnClickOutside(MainSidebarRef, () => {
+    if (windowSize.width <= 950) {
+      setShowSidebar(false);
+    }
+  });
+
+  const linkClickHandler = () => {
+    if (windowSize.width <= 950) {
+      setShowSidebar(false);
+    }
+  };
+
+  return (
+    <div ref={MainSidebarRef} className="expanded-sidebar">
+      <header className="sidebar-header">
+        <section>
+          <button className="nav-menu" onClick={() => setShowSidebar(false)}>
+            <span className="material-icons nav-menu-icon">close</span>
```

Comment:
If the className is given as `material-icons` then why it is a text?

Reply:
We give material icons class and then give text of name of icon we need - this is based on Google action icons